### PR TITLE
Implement debugging hook for consumeInitialDOM

### PR DESCRIPTION
**Pull request by choumx**
Allows callers to inspect/print the `HydrateableNode` that used during worker bootstrapping.

### CODE REVIEW

**Comment by kristoferbaxter** [comment]
@choumx – There is a conflict now with merged changes from Justin's work. Can you take a look and we can re-review?

**Comment by choumx** [comment]
On it. Not trivial to rebase but I have an idea of how to do it.

**Comment by choumx** [comment]
Done, PTAL. Instead of deserializing the HydrateableNode, I create another "readable" version from the `baseElement`.

**Comment by kristoferbaxter** [comment]
Looking now.

File: src/main-thread/worker.ts [hunk 1]
```diff
@@ -76,7 +76,11 @@ export function createWorker(
       ${authorScript}
     }).call(WorkerThread.workerDOM);
 //# sourceURL=${encodeURI(authorScriptURL)}`;
-  return new Worker(URL.createObjectURL(new Blob([code])));
+  const worker = new Worker(URL.createObjectURL(new Blob([code])));
+  if (callbacks && callbacks.onCreateWorker) {
```
Comment (kristoferbaxter) [inline src/main-thread/worker.ts]: Here's another place where the debug build varies from the production build. Let's make sure these additions are only present in the debug build (in a separate PR).

**Pull request merged**